### ISSUE TITLE
Documentation on kopia authentication is unclear and confusing

### ISSUE THREAD

**Issue by nakermann1973** [kopia/kopia#3783]
When running a repository server (in docker at least), there are multiple user accounts and password which should be configured.   Most of them are self-explanatory (e.g. repository password and repository users), but the http and server control users are problematic:

When starting the server (`kopia server start`), a username and password for accessing the HTTP web interface is specified (note, this appears to be unrelated to server or repository usernames and passwords, and is a separate username just for the UI - is this correct?)

Note that `KOPIA_SERVER_USERNAME` and `KOPIA_SERVER_PASSWORD` can be used to specify this user/pass

```
      --server-username="kopia"  HTTP server username (basic auth) ($KOPIA_SERVER_USERNAME)
      --server-password=SERVER-PASSWORD  
                                 HTTP server password (basic auth) ($KOPIA_SERVER_PASSWORD)
```

If I want to interact with the server from the CLI (for example to run `kopia server status`), I need to add a few arguments to  the `kopia server start` command (which can use `KOPIA_SERVER_CONTROL_USER` and `KOPIA_SERVER_CONTROL_PASSWORD` environment variables)

```
      --server-control-username="server-control"                                                                                                                                                      
                                 Server control username ($KOPIA_SERVER_CONTROL_USER)                                                                                                                 
      --server-control-password=PASSWORD                                                                                                                                                              
                                 Server control password ($KOPIA_SERVER_CONTROL_PASSWORD)                                                                                                             
```

When I run `kopia server status` though, different environment variables are used (e.g. not `KOPIA_SERVER_CONTROL_USER`, but instead `KOPIA_SERVER_USERNAME`):

```
      --server-control-username=SERVER-CONTROL-USERNAME  
                              Server control username ($KOPIA_SERVER_USERNAME)
      --server-control-password=PASSWORD  
                              Server control password ($KOPIA_SERVER_PASSWORD)
```

This makes it very difficult to run a docker container which reads the environment variables from a secret store, because the variables are used for different things by different commands.  This looks like a bug to me.

**Comment by danjl1100** [comment]
Would a PR be accepted to harmonize the cli arguments and env vars for the "cli_server" and "cli_server_start" commands?

Initially this seemed small enough for a drive-by PR, but it seems there are a few details for discussion.

### UX (Goal State for New Users)

- Issue 1: Environment vars corresponding to `--server-control-{username,password}` are inconsistent between `server` and `server start` CLI commands
    - Option A: Update `server` env vars to match `server start`,    updating from KOPIA_SERVER_PASSWORD to KOPIA_SERVER_*CONTROL*_PASSWORD. Same for "username", per chosen solution for Issue 2.
    - Option B: Leave as-is
- Issue 2: `KOPIA_SERVER_CONTROL_USER` does not use the same text as `--server-control-username`
    - Option A: Update env var from KOPIA_SERVER_CONTROL_USER to KOPIA_SERVER_CONTROL_USER*NAME*
    - Option B: Leave as-is
- Issue 3: (New, not in original issue) For new users, it's not entirely clear what `--server-{username,password}` does as opposed to `--server-control-{username,password}`.
    - Option A: Rename --server-password to --server-**ui**-password (env KOPIA_SERVER_*UI*_PASSWORD).  Same for "username", per chosen solution for Issue 2.
    - Option B: Leave as-is

To me 1.A and 2.A seem preferable.  I'm on the fence about adding 3.A into this change too, but it might make sense (depending on Issue 4 below) to help further clarify the meaning for new users.

### Semver (Upgrade Path for Existing Users)
As far as semver  / migration for existing users, which approach is preferred by the project?
- Issue 4: Any change to CLI arguments or environment variables (Issues 1-3 above) breaks existing user scripts
    - Option A: Major version bump (v0.19.0?) updates the CLI argument and environment variable naming. Migration documentation and CLI errors will tell users to update their scripts.
    - Option B: Make the changes backwards-compatible, allowing either old names OR new names. Error if both names are specified.
    - Option C: Same as Option B, and also print a deprecation warning for any old names encountered:
        ```
        WARN: environment variable KOPIA_SERVER_USERNAME is deprecated in favor of
        KOPIA_SERVER_CONTROL_USERNAME, and may error in a future version of kopia
        ```

My preference is 4.C, though it adds slightly more complexity over 4.B.

---

Looking into the code, the CLI argument / environment variable mapping seems well contained to two functions:
- [`func serverStartOptions`](https://github.com/kopia/kopia/blob/ef016506657fcdf2882ca162cf78cc735dca892c/cli/command_server_start.go#L139) in `cli/command_server_start.go`
- [`func serverAPIClientOptions`](https://github.com/kopia/kopia/blob/ef016506657fcdf2882ca162cf78cc735dca892c/cli/command_server.go#L77) in `cli/command_server.go`

**Comment by jkowalski** [comment]
Yes, we should fix those inconsistencies, but it's really important to avoid breaking the users.

That means:

- legacy flags should continue to work (but be hidden, not advertised)
- we should update documentation to use the recommended flag/environment names.
- we should have tests for both old/new flags to make sure they continue to work and we don't accidentally break them in the future 
- we should warn people using deprecated flags (we have `deprecatedFlag()` for this)
- there should perhaps be a "strict" mode where using deprecated flags triggers a failure, not a warning.

In other words:

Issues 1&2 - we should support both `KOPIA_SERVER_CONTROL_USERNAME` and `KOPIA_SERVER_CONTROL_PASSWORD` (primary, in the documentation), but maintain compatibility by also supporting old names. This can be achieved by declaring two flags that refer to the same variable in code.
We already have some legacy there:

```
	cmd.Flag("server-control-username", "Server control username").Envar(svc.EnvName("KOPIA_SERVER_USERNAME")).StringVar(&c.serverUsername)
	cmd.Flag("server-control-password", "Server control password").PlaceHolder("PASSWORD").Envar(svc.EnvName("KOPIA_SERVER_PASSWORD")).StringVar(&c.serverPassword)

	// aliases for backwards compat
	cmd.Flag("server-username", "Server control username").Hidden().StringVar(&c.serverUsername)
	cmd.Flag("server-password", "Server control password").Hidden().StringVar(&c.serverPassword)
```

Issue 3: Let's do it - I do like the UI to be explicitly mentioned.

Issue 4: Option 4.C is the way to go.

I'd be happy to review/merge PR that implements all of this, perhaps even 2 PRs - one for server control and one for the UI.

**Comment by danjl1100** [comment]
Thank you for the quick response!

I'll start looking into the tests and docs for implementing the changes.  2 PRs sounds like a good idea to me, maybe even 3 for strict mode which would affect the existing "backwards compat" aliases too. 

For the example you quoted of existing "aliases for backwards compat":
In order to error when both old and new names are used (part of 4.C, really 4.B), would there need to be 2 copies of the fields in the `commandServer` struct? (e.g. `serverUsername` and `serverUsernameCompat1`) I'm not sure how else to preserve all the user input data to raise the error in `func serverAPIClientOptions`.

**Comment by AlexpFr** [comment]
Hi everyone,

I just wanted to bump this issue as I recently spent a significant amount of time trying to properly configure a Kopia Docker container. As a new user, the current overlap between environment variables (specifically KOPIA_SERVER_USERNAME acting differently for the Web UI vs. the CLI Client) was quite confusing and led to a lot of trial and error.

I see that PR #4435 addresses this by clarifying the arguments. Is there any update on the status of this PR or a timeline for when we might see these changes merged?

Separating _UI_ and _CONTROL_ credentials explicitly would be a massive quality-of-life improvement for new deployments.

Thanks for the hard work on the project!

**Comment by danjl1100** [comment]
Hello!
Glad to hear that this fix will be helpful to you too.

I lost track of that PR, need to figure out the status. Other than rebasing on the latest version, I seem to recall the implementation was mostly there.
It could be stuck on needing the corresponding documentation updates, too. That's something I haven't looked into yet.

**Comment by AlexpFr** [comment]
Thanks a lot for your reply!

I'm currently working on setting up the Kopia server and repository using best practices from both Docker and Kopia, and starting from scratch with Kopia isn’t that easy.

Improvements or clarifications like this around commands and environment variables would be really helpful for new users and for Docker setups.

**Comment by danjl1100** [comment]
I found some time to rebase PR #4435 against v0.22.3 for the new compat-tests. Let me know if other updates are needed to bring the PR closer to being accepted.

**Comment by Ttfgggf** [comment]
#5496

**Comment by DiegoBM** [comment]
Will this ever be merged? It has been active for more than 2 years. Also while at it, could it be possible to support password_files in order to leverage docker secrets?
